tests/plugins/test_fetch_worker.py

```diff
@@ -0,0 +1,162 @@
+"""
+Copyright (c) 2017 Red Hat, Inc
+All rights reserved.
+
+This software may be modified and distributed under the terms
+of the BSD license. See the LICENSE file for details.
+"""
+
+from __future__ import print_function, unicode_literals
+
+import os
+
+from flexmock import flexmock
+
+from atomic_reactor.core import DockerTasker
+from atomic_reactor.constants import PLUGIN_FETCH_WORKER_METADATA_KEY
+from atomic_reactor.build import BuildResult
+from atomic_reactor.inner import DockerBuildWorkflow
+from atomic_reactor.plugin import PostBuildPluginsRunner
+from atomic_reactor.util import ImageName
+
+from atomic_reactor.plugins.build_orchestrate_build import (WorkerBuildInfo, ClusterInfo,
+                                                            OrchestrateBuildPlugin)
+
+from tests.constants import MOCK_SOURCE, TEST_IMAGE, INPUT_IMAGE
+from tests.docker_mock import mock_docker
+import pytest
+
+
+class mock_ConfigMapResponse(object):
+    def __init__(self, data):
+        self.data = data
+
+    def get_json_by_key(self, key):
+        return self.data[key]
+
+
+class mockOSBS(object):
+    def __init__(self, config_map):
+        self.config_map = config_map
+
+    def get_config_map(self, name):
+        return mock_ConfigMapResponse(self.config_map[name])
+
+
+class MockSource(object):
+
+    def __init__(self, tmpdir):
+        tmpdir = str(tmpdir)
+        self.dockerfile_path = os.path.join(tmpdir, 'Dockerfile')
+        self.path = tmpdir
+
+    def get_dockerfile_path(self):
+        return self.dockerfile_path, self.path
+
+
+class MockInsideBuilder(object):
+
+    def __init__(self):
+        mock_docker()
+        self.tasker = DockerTasker()
+        self.base_image = ImageName(repo='fedora', tag='25')
+        self.image_id = 'image_id'
+        self.image = INPUT_IMAGE
+        self.df_path = 'df_path'
+        self.df_dir = 'df_dir'
+
+        def simplegen(x, y):
+            yield "some\u2018".encode('utf-8')
+        flexmock(self.tasker, build_image_from_path=simplegen)
+
+    def get_built_image_info(self):
+        return {'Id': 'some'}
+
+    def inspect_built_image(self):
+        return None
+
+    def ensure_not_built(self):
+        pass
+
+
+def mock_workflow(tmpdir):
+    workflow = DockerBuildWorkflow(MOCK_SOURCE, TEST_IMAGE)
+    setattr(workflow, 'builder', MockInsideBuilder())
+    setattr(workflow, 'source', MockSource(tmpdir))
+    setattr(workflow.builder, 'source', MockSource(tmpdir))
+
+    return workflow
+
+
+@pytest.mark.parametrize('fragment_key', ['metadata.json', None])
+def test_fetch_worker_plugin(tmpdir, fragment_key):
+    workflow = mock_workflow(tmpdir)
+
+    annotations = {
+        'worker-builds': {
+            'x86_64': {
+                'build': {
+                    'build-name': 'build-1-x64_64',
+                },
+                'metadata_fragment': 'configmap/build-1-x86_64-dm',
```

Comment:
Only mentioning this in case it's a typo: koji_upload will append "-md" (for 'metadata'), not '-dm'.
https://github.com/projectatomic/atomic-reactor/pull/707/files#diff-5e7041655266b243b2a76e9c928c99b5R624